### PR TITLE
Make all openshift/assisted-service presubmit jobs run if data directory was changed

### PR DESCRIPTION
**Pull request by YuviGold**
In addition, fix images job to always run.

This would run everything for our OCP bumps.
/cc @osherdp 
/cc @gamli75

### CODE REVIEW

File: ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml [hunk 1]
```diff
@@ -62,7 +62,7 @@ presubmits:
       pj-rehearse.openshift.io/can-be-rehearsed: "true"
     name: pull-ci-openshift-assisted-service-master-e2e-metal-assisted
     rerun_command: /test e2e-metal-assisted
-    run_if_changed: ^(cmd/.*|deploy/.*|hack/.*|internal/.*|pkg/.*|tools/.*|Dockerfile\..*|Makefile|go\.mod|go\.sum|swagger.yaml)$
+    run_if_changed: ^(cmd/.*|data/.*|deploy/.*|hack/.*|internal/.*|pkg/.*|tools/.*|Dockerfile\..*|Makefile|go\.mod|go\.sum|swagger.yaml)$
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:65]: anything needs to be changed on ``ocm-2.3``?

**Review by osherdp — CHANGES_REQUESTED** [review]

**Comment by osherdp** [comment]
/hold
(just to be on the safe side)

File: ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml (hunk 1 above)
Comment (YuviGold) [reply ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:65]: no. the multi versions is cloud related

Comment (osherdp) [reply ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:65]: but it has also ``default_hw_requirements.json``

Comment (YuviGold) [reply ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:65]: those variables are handled via env variables. Those are just the defaults that are used in the cloud :)

Comment (YuviGold) [reply ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:65]: We shouldn't backport data values

**Review by osherdp — APPROVED** [review]

File: ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml (hunk 1 above)
Comment (osherdp) [reply ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:65]: ok, if it's not relevant to the operator then it's ok

**Pull request merged**